doc/md/guides/migration-tools/golang-migrate.md

```diff
@@ -0,0 +1,218 @@
+```
+
+3\. Close the shell and run `golang-migrate` to run all migrations on our new database:
+
+```text
+migrate -source file://migrations -database 'mysql://root:pass@tcp(localhost:3306)/migrate-current' up
+```
+All migrations are executed successfully:
+```text
+1/u init (35.601678ms)
+```
+
+4\. Next, use Atlas's `schema inspect` command to write a file named `schema.hcl` with an HCL
+representation of your migration directory. Notice that we exclude the `schema_migrations` table
+which contains `golang-migrate`'s revision history. 
+
+```text
+atlas schema inspect -u mysql://root:pass@localhost:3306/migrate-current --exclude '*.schema_migrations' > schema.hcl
+```
+
+Observe that a new file named `schema.hcl` was created:
+
+```hcl
+table "users" {
+  schema = schema.migrate-current
+  column "id" {
+    null = false
+    type = int
+  }
+  column "name" {
+    null = true
+    type = varchar(100)
+  }
+  primary_key {
+    columns = [column.id]
+  }
+}
+schema "migrate-current" {
+  charset = "utf8mb4"
+  collate = "utf8mb4_0900_ai_ci"
+}
+```
+
+### Plan a new migration
+
+Next, let's modify the desired state of our database by modifying the `schema.hcl` file 
+to add a new table `blog_posts` that has a foreign key pointing to the existing `users`
+table:
+
+```hcl
+table "blog_posts" {
+  schema = schema.migrate-current
+  column "id" {
+    type = int
+  }
+  column "title" {
+    type = varchar(255)
+  }
+  column "body" {
+    type = text
+  }
+  column "author_id" {
+    type = int
+  }
+  foreign_key "blog_author_fk" {
+    columns = [column.author_id]
+    ref_columns = [table.users.column.id]
+  }
+}
+```
+
+Now, let's use Atlas's `migrate diff` command to plan a migration from the current state
+as it exists in the migrations directory to the desired state that is defined by the `schema.hcl`
+file:
+
+```text
+atlas migrate diff --dir file://migrations  --dev-url mysql://root:pass@localhost:3306/dev --to file://schema.hcl --dir-format golang-migrate add_blog_posts 
+```
+
+Notice that we used the `dir-format` flag to specify that we're using `golang-migrate` as the directory format.
+
+Hooray! Two new files were created in the migrations directory:
+```text {5-6}
+.
+├── migrations
+│ ├── 1_init.down.sql
+│ ├── 1_init.up.sql
+│ ├── 20220922123326_add_blog_posts.down.sql
+│ ├── 20220922123326_add_blog_posts.up.sql
+│ └── atlas.sum
+└── schema.hcl
+```
+An up migration:
+```sql
+-- create "blog_posts" table
+CREATE TABLE `blog_posts` (`id` int NOT NULL, `title` varchar(255) NOT NULL, `body` text NOT NULL, `author_id` int NOT NULL, INDEX `blog_author_fk` (`author_id`), CONSTRAINT `blog_author_fk` FOREIGN KEY (`author_id`) REFERENCES `users` (`id`) ON UPDATE NO ACTION ON DELETE NO ACTION) CHARSET utf8mb4 COLLATE utf8mb4_0900_ai_ci;
+```
+And a down migration:
+```sql
+-- reverse: create "blog_posts" table
+DROP TABLE `blog_posts`;
+```
+
+### Alternative: use an existing database as the desired state
```

Comment:
As a reader, if I see the part about the HCL and don't wanna learn it, I would stop reading. Maybe add a small info box linking here at that place? If someone starts to work with Atlas, they will later adapt HCL, I am sure of it.